config/set/php52.php

```diff
@@ -22,4 +24,11 @@
                 new RemoveFuncCallArg('ldap_first_attribute', 2),
             ]),
         ]]);
+
+    $services->set(ChangePackageVersionComposerRector::class)
+        ->call('configure', [[
+            ChangePackageVersionComposerRector::PACKAGES_AND_VERSIONS => ValueObjectInliner::inline([
+                new PackageAndVersion('php', '^5.2'),
+            ]),
```

Comment:
Btw, do we have a test to avoid lowering PHP version? 

So we can avoid downgrading bugs like these:

```diff
 {
-   "php": "^7.3"
+   "php": "^5.2"
 }
```


If not, this PR should contain this test too

Reply:
You mean if it compare version strings and change only if target version is higher than original one? No those test don’t exist and it is good because you should use this rules also for downgrading

Reply:
> You mean if it compare version strings and change only if target version is higher than original one?

Yes :+1:

Reply:
> it is good because you should use this rules also for downgrading

We have downgrading sets, haven't we?

Reply:
Yes, that's why we shouldn't check if new version is higher as previous

Reply:
Not sure if I follow the discussion correctly: If my composer requires `^7.4` and I run the `PHP 5.6` upgrade set, that this will now change my composer to require `^5.6`?

Reply:
I hope not :)
We need a test case to avoid it.

Reply:
Maybe there are some misunderstandings from me how rector is used with php sets in a wild. Here are my assumptions:
If app / library requires php ^7.3, it run:
- only set php7.3 
- or more php sets - let's say php7.1, php7.2, php7.3. Set for actual version is always the last running.

Same steps are used in rector itself in rector.php. 

If your lib / app using different version of php or some range, you can set it in your rector config - like it is in this merge request for rector https://github.com/rectorphp/rector/pull/5398/files#diff-d933a2ad57daa43419b483d3f1ddedaec7aa44933007fead9f9437390f596f4aR94

Reply:
For what it's worth: This is how I used it: Our application runs on PHP 7.4 and requires that in composer. I just discovered Rector and applied sets starting from PHP 5.2, one by one till 7.1 (need to do some more work to do all the upgrades up until 7.4). Turned out that Rector found lots of things that we were doing "the old way". In any case, our composer PHP version should remain 7.4.

Reply:
I understand, so you would need just add one setting in your rector.php to keep it unchanged:
```
$services->set(ChangePackageVersionComposerRector::class)
        ->call('configure', [[
            ChangePackageVersionComposerRector::PACKAGES_AND_VERSIONS => ValueObjectInliner::inline([
                new PackageAndVersion('php', '^7.4'),
            ]),
        ]]);
```

Reply:
@lulco Adding work to our users is not a Rector way. This is our responsibility to handle.

Reply:
Can't this automatically be read from `$parameters->set(Option::PHP_VERSION_FEATURES, PhpVersion::PHP_74);`?